apps/discovery_api/lib/discovery_api_web/controllers/multiple_metadata_controller.ex

```diff
@@ -51,6 +53,41 @@ defmodule DiscoveryApiWeb.MultipleMetadataController do
       reraise e, __STACKTRACE__
   end
 
+  def fetch_table_info(conn, _params) do
+    user_id = get_user_id(conn)
+
+    filtered_models =
+      case TableInfoCache.get(user_id) do
+        nil ->
+          remove_unauthorized_models(conn, Model.get_all())
+          |> get_filtered_table_info()
+          |> TableInfoCache.put(user_id)
+
+        filtered_models ->
+          filtered_models
+      end
+
+    render(
+      conn,
+      :fetch_table_info,
+      models: filtered_models
+    )
+  end
```

Comment:
Is there any merit to the idea of putting tableau-related stuff into its own controller?  Not saying you need to make a bunch of changes - just a thought.

Reply:
done